api/model/api.go

```diff
@@ -1049,3 +1058,30 @@ func (ls ListAPIOptions) Params() url.Values {
 
 	return u
 }
+
+func (in *ConfigDataType) DeepCopyInto(out *ConfigDataType) {
+	// controller-gen cannot handle the interface{} type of an aliased Unstructured,
+	// thus we write our own DeepCopyInto function.
+	if out != nil {
+		casted := unstructured.Unstructured(in.Unstructured)
+		deepCopy := casted.DeepCopy()
+		out.Object = deepCopy.Object
+	}
+}
+
+func (in *ConfigDataType) UnmarshalJSON(data []byte) error {
+	m := make(map[string]interface{})
+	if err := json.Unmarshal(data, &m); err != nil {
+		return err
+	}
+
+	in.Object = m
+
+	return nil
+}
+
+func (in *ConfigDataType) MarshalJSON() (data []byte, err error) {
+	m := in.Object
+
+	return json.Marshal(m)
```

Comment:
Does `in.Object` need to be assigned to a new `m` variable? If not, we might return as:
```go
return json.Marshal(in.Object)
```

Reply:
Nice spot. Fixed in `5f25444`